expression/builtin_json_vec.go

```diff
@@ -91,11 +93,49 @@ func (b *builtinJSONSetSig) vecEvalJSON(input *chunk.Chunk, result *chunk.Column
 }
 
 func (b *builtinJSONObjectSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinJSONObjectSig) vecEvalJSON(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	nr := input.NumRows()
+	if len(b.args)&1 == 1 {
+		err := ErrIncorrectParameterCount.GenWithStackByArgs(ast.JSONObject)
+		return err
+	}
+
+	jsons := make([]map[string]interface{}, nr)
+	for i := 0; i < nr; i++ {
+		jsons[i] = make(map[string]interface{}, len(b.args)>>1)
+	}
+
+	result.ReserveJSON(nr)
+	for i := 0; i < len(b.args); i++ {
+		if i&1 == 1 {
+			keyCol := input.Column(i - 1)
+			valueCol := input.Column(i)
```

Comment:
We should invoke its children to get the `keyCol` and `valueCol` for example `b.args[i-1].VecEvalString(...)` instead of using the input data.
The reason why it passed the local test is that in the local test the expression tree is relatively simple where the children of this function all are `Column`.

Reply:
ok, I get it. I will fix it.